### PR TITLE
Table Menu

### PR DESCRIPTION
**Pull request by kfk12688**
render the table with virtual table component

### CODE REVIEW

**Comment by kfk12688** [comment]
sir, i didn't totch dataindex component i ll work it monday

**Comment by Sharavanth** [comment]
OK. Please inform me once you have completed

File: src/DataRoute/Table.js [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+      <div>
+        <Alert
+          intent="danger"
+          header="Delete Item"
+          onConfirm={this.deleteItem}
+          isOpen={deleteItemFlag}
+          onCancel={this.removeDeleteItemFlag}
+          description="Are you sure you want to delete this Item?"
+        />
+        {isDefined(contentValues.userSchema) &&
```
Comment (Sharavanth) [inline src/DataRoute/Table.js]: 1. on user click only the spec needs to be passed

File: src/DataRoute/Table.js [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+          <DataForm
+            onSubmit={addItem}
+            isOpen={editItemFlag}
+            spec={contentValues.userSchema}
+            onClose={this.removeEditItemFlag}
+          />}
+        <Row />
```
Comment (Sharavanth) [inline src/DataRoute/Table.js]: What is this for..... Add a comment at least

File: src/DataRoute/Table.js [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+                <Button
+                  icon="trash"
+                  intent="warning"
+                  onClick={this.setDeleteItemFlag(id)}
+                >
+                  Delete
+                </Button>}
+            />
+          </VirtualTable>
+          <Cards>
```
Comment (Sharavanth) [inline src/DataRoute/Table.js]: `<Column>` is missing here

File: src/DataRoute/Table.js [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+            <Label>
```
Comment (Sharavanth) [inline src/DataRoute/Table.js]: use `text` prop here

File: src/DataRoute/Table.js [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+    );
+  }
+
+  componentDidMount() {
+    this.toaster = Toaster.create({ position: POSITIONS.TOP });
+  }
+
+  getColumns(schema) {
+    return map(
+      scheme => <Col label={scheme.displayName} render={this.renderCallback(scheme)} />,
```
Comment (Sharavanth) [inline src/DataRoute/Table.js]: Please make sure to give proper names for functions/ variables
change `renderCallback `to `renderCellCallback`

File: src/DataRoute/Table.js [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+    const { fieldName, fieldType } = scheme;
+
+    const currentField = {
+      text: items[id][fieldName].toString(),
+      boolean: Boolean(items[id][fieldName]),
+      integer: parseInt("0", items[id][fieldName]),
+      decimal: parseFloat(items[id][fieldName]),
+      longtext: items[id][fieldName]
+        .split("\n")
+        .reduce((prevVal, addBr) => prevVal + addBr, 0),
```
Comment (Sharavanth) [inline src/DataRoute/Table.js]: I think you do not understand the reduce func......

File: src/DataRoute/Table.js [hunk 7]
```diff
@@ -0,0 +1,162 @@
...
+  }
+
+  renderCallback = scheme => id => {
+    const { items } = this.props.store;
+    const { fieldName, fieldType } = scheme;
+
+    const currentField = {
+      text: items[id][fieldName].toString(),
+      boolean: Boolean(items[id][fieldName]),
+      integer: parseInt("0", items[id][fieldName]),
```
Comment (Sharavanth) [inline src/DataRoute/Table.js]: wrong usage of parseInt

File: src/DataRoute/Table.js [hunk 8]
```diff
@@ -0,0 +1,162 @@
...
+      longtext: items[id][fieldName]
+        .split("\n")
+        .reduce((prevVal, addBr) => prevVal + addBr, 0),
+      list: items[id][fieldName].toString(),
+      date: moment(items[id][fieldName]).format("DD-MM-YYYY"),
+      month: moment(items[id][fieldName]).format("MMM-YYY"),
+      time: moment(items[id][fieldName]).format("h:mm:ss a"),
+      reference: !!items[id][fieldName] && items[id][fieldName].label,
+      image: <img src={items[id][fieldName]} alt="thumbnail" height={60} />,
+      schema: (items[id][fieldName] = {}),
```
Comment (Sharavanth) [inline src/DataRoute/Table.js]: Asked you to remove this... Ryt???

File: src/DataRoute/Table.js [hunk 9]
```diff
@@ -0,0 +1,162 @@
...
+
+    return currentField(fieldType);
+  };
+
+  deleteItem = () => {
+    const { deleteItemId } = this.state;
+    this.setState({ deleteItemFlag: false });
+    this.props
+      .deleteItem(this._tableName, deleteItemId)
+      .then(this.context.router.push("/tables"));
```
Comment (Sharavanth) [inline src/DataRoute/Table.js]: This line will not work....... Please understand and then code

File: src/DataRoute/Table.js [hunk 10]
```diff
@@ -0,0 +1,162 @@
...
+  };
+
+  setEditItemFlag = e => this.setState({ editItemFlag: true });
+  setDeleteItemFlag = (e, id) =>
+    this.setState({ deleteItemFlag: true, deleteItemId: id });
+  removEditItemFlag = e => this.setState({ editItemFlag: false });
```
Comment (Sharavanth) [inline src/DataRoute/Table.js]: Not used anywhere......

File: src/DataRoute/Table.js [hunk 11]
```diff
@@ -0,0 +1,162 @@
...
+      editItemFlag: false,
+    };
+    this._tableName = props.tableName;
+  }
+
+  render() {
+    const { items, userSchema } = this.props.store;
+    const { addItem, tableName } = this.props;
+    const { editItemFlag, deleteItemFlag } = this.state;
+    const contentValues = getContentValues(this.props.store[tableName]);
```
Comment (Sharavanth) [inline src/DataRoute/Table.js]: Is this function neccesary??

File: src/DataRoute/Table.js (hunk 9 above)
Comment (kfk12688) [reply src/DataRoute/Table.js]: sir, i have a doubt in how to push this url

**Pull request closed without merging**